keps/sig-storage/695-skip-permission-change/README.md

```diff
@@ -105,6 +106,37 @@ type PodSecurityContext struct {
 }
 ```
 
+#### Handling of volume types that apply gid as a mount option
+
+Handling of volume types that use fsGroup as a mount option is out of scope for 1.20 beta, but following section still describes the general mechanism that will be used.
+
+We propose following API change to `CSIDriver` type to allow drivers to declare support for applying fsGroup during mount time.
+
+```go
+const (
+    ReadWriteOnceWithFSTypeFSGroupPolicy FSGroupPolicy = "ReadWriteOnceWithFSType"
+    FileFSGroupPolicy FSGroupPolicy = "File"
+
+    // OnMountFSGroupPolicy indicates that CSI driver supports changing volume ownership via
+    // mount flags and hence fsgroup of pod should be made available to CSI driver in nodePublish
+    // and nodeStage CSI RPC calls.
+    OnMountFSGroupPolicy FSGroupPolicy = "Mount" <--- new change
+
+    NoneFSGroupPolicy FSGroupPolicy = "None"
+)
+```
+
+If `CSIDriver.Spec.FSGroupPolicy` is set to `Mount` then pod's fsGroup will be supplied to the CSI driver in nodeStage/nodePublish driver RPC
+calls. `FSGroupChangePolicy` of pod will not have any effect because no recursive ownership and permission change is necessary.
+The exact mechanism of supplying `fsGroup` to CSI driver is still being worked out and is not part of 1.20 beta milestone for this feature.
+
+We are currently considering three alternatives for supplying fsGroup to `NodeStage` and `NodePublish` RPC calls:
```

Comment:
Do we have pod fsgroup at NodeStage time? What if multiple pods use 2 different fsgroup?

Reply:
+1 
Does this mean that we need the driver to expose some kind of capability which describes if or how it will apply the mount policy?

That consideration seems to make updating the CSI spec the right way to go forward with this.

Reply:
Currently we do not have `fsGroup` available in `MountDevice` function(which calls `NodeStage`) but I think that is trivially fixable to make `fsGroup` available (let me know if I understood this correctly).

As for what happens if more than one pod use different fsGroup and same volume - I don't have a good answer. There could be two cases:
1. If volume is already mounted on same node and can not be remounted with different fsGroup then mount for second pod should fail I think.
2. In some cases I think - if a fuse daemon or something similar is squashing and mapping the gid of process writing the file to some underlying value then(bit like NFS root squash), it might be possible to mount same volume with more than one gid on two different nodes. 

I think k8s has gone with quite loosely defined APIs for fsGroup and seLinux contexts and now it is somewhat hard to tighten them. If we had to redo - then we would probably want to stop creation of pods that reuse a volume with different fsGroup concurrently. I think this would apply to subpaths too. It would keep our APIs cleaner and make error handling easy to reason about. I don't know if we can take this path now though... :(

Reply:
Can you document what the behavior will be if this happens? Either here or the CSIDriver kep